engines/saga/image.cpp

```diff
@@ -43,11 +43,44 @@ static int granulate(int value, int granularity) {
 	}
 }
 
+bool SagaEngine::decodeBGImageMask(const ByteArray &imageData, ByteArray &outputBuffer, int *w, int *h, bool flip) {
+	if (isAGA() || isECS()) {
+		if (imageData.size() < 160 * 137 + 64)
+			return false;
+		*w = 320;
+		*h = 137;
+		outputBuffer.resize(320*137);
+
+		// First read types
+		for (int i = 0; i < 160*137; i++) {
+			outputBuffer[2 * i] = (imageData[i] << 4) | 0xf;
+			outputBuffer[2 * i + 1] = (imageData[i] << 4) | 0xf;
+		}
+
+		// Now instead of storing depth amiga variant stores precomputed mask for every
+		// depth. Obviously not every set of precomputed masks is valid but we assume
+		// that it is and suffer graphical glitches when it isn't
```

Comment:
What do you mean with graphical glitches here? Do we have a case where the above is not a valid assumption?

Reply:
It never happened so far. I will clarify it